lib/reader/src/parser.rs

```diff
@@ -1653,4 +1649,39 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn instruction_encoding() {
+        use cretonne::ir::Inst;
+        use cretonne::entity_map::EntityRef;
+
+        let test_file = parse_test("isa riscv supports_m
```

Comment:
This is fine, but note that it would probably be easier to write this as a file test under `filetests/parser/`. This would let you round-trip these parser changes fairly simply.